build/commands/lib/util.js

```diff
@@ -277,31 +296,6 @@ const util = {
       }
     }
 
-    if (process.platform === 'darwin') {
-      // Copy proper mac app icon for channel to chrome/app/theme/mac/app.icns.
-      // Each channel's app icons are stored in brave/app/theme/$channel/app.icns.
-      // With this copying, we don't need to modify chrome/BUILD.gn for this.
-      const iconSource = path.join(braveAppDir, 'theme', 'brave', 'mac', config.channel, 'app.icns')
-      const iconDest = path.join(chromeAppDir, 'theme', 'brave', 'mac', 'app.icns')
-      if (!fs.existsSync(iconDest) ||
-          util.calculateFileChecksum(iconSource) != util.calculateFileChecksum(iconDest)) {
-        console.log('copy app icon')
-        fs.copySync(iconSource, iconDest)
-      }
-
-      // Copy branding file
-      let branding_file_name = 'BRANDING'
-      if (config.channel)
-        branding_file_name = branding_file_name + '.' + config.channel
-
-      const brandingSource = path.join(braveAppDir, 'theme', 'brave', branding_file_name)
-      const brandingDest = path.join(chromeAppDir, 'theme', 'brave', 'BRANDING')
-      if (!fs.existsSync(brandingDest) ||
-          util.calculateFileChecksum(brandingSource) != util.calculateFileChecksum(brandingDest)) {
-        console.log('copy branding file')
-        fs.copySync(brandingSource, brandingDest)
-      }
-    }
```

Comment:
this was always rewriting `BRANDING` and `icns` files, which triggered unnecessary rebuilds during `create_dist` in non-development channels.

Reply:
running `npm run test ...` without first running `npm run build ...` fails now
```
ninja: Entering directory `/Volumes/brave/src/out/Component'
ninja: error: '../../components/vector_icons/brave/product_refresh.icon', needed by 'gen/components/vector_icons/vector_icons.cc', missing and no known rule to make it
```

Reply:
> running `npm run test ...` without first running `npm run build ...` fails now
> 
> ```
> ninja: Entering directory `/Volumes/brave/src/out/Component'
> ninja: error: '../../components/vector_icons/brave/product_refresh.icon', needed by 'gen/components/vector_icons/vector_icons.cc', missing and no known rule to make it
> ```

yes, that was always the case. `npm run test` also never calls `gn gen`, so you had to call `npm run build` at least once to prepare everything. I usually do `npm run build -- --target=base` if I want to immediately run some test on a fresh checkout.